### PR TITLE
Bug/fp 1798 external nav link (still) not open in new window

### PR DESCRIPTION
**Pull request by wesleyboar**
## Overview

Correct logic that should make CMS nav menu links to external pages open in new window, but was not working.

## Related

- [FP-1798](https://jira.tacc.utexas.edu/browse/FP-1798)
- fixes https://github.com/TACC/Core-CMS/pull/453

## Changes

- Use correct conditional logic to get redirect_uri from a cms_menu child.
- Check all links (header and footer), not just main content links.

## Testing & UI

See https://github.com/TACC/Core-CMS/pull/453 at:
- https://dev.cep.tacc.utexas.edu/test/fp-1451-page-content-links (content links and menu link it mentions)
- https://pprd.brainmap.tacc.utexas.edu/ (menu link "Tools ▸ Sleuth 3.0.4")

### CODE REVIEW

File: taccsite_cms/templatetags/tacc_uri_shortcuts.py [hunk 1]
```diff
@@ -111,7 +111,8 @@ def get_menu_uri(menu_item):
             {{ child.get_menu_title }}</a>
             # <a href="https://otherwebsite.com/some-page">some page there</a>
     """
-    if hasattr(menu_item.attr, 'redirect_url'):
```
Comment (wesleyboar) [inline taccsite_cms/templatetags/tacc_uri_shortcuts.py:114]: This python was wrong. [I should use `in` operator to test for members of a dictionary.](https://stackoverflow.com/a/10724850/11817077) I do not know how this was tested successfully in https://github.com/TACC/Core-CMS/pull/453. I would bet I had an old working snippet solution active (and working) at the same time I was testing the new branch.

File: taccsite_cms/templates/nav_cms.html [hunk 2]
```diff
@@ -12,7 +12,5 @@
 <script type="module">
   import findLinksAndSetTargets from '/static/site_cms/js/modules/setTargetForExternalLinks.js';
 
-  const scopeElement = document.getElementById('cms-content');
-
-  findLinksAndSetTargets({ scopeElement });
+  findLinksAndSetTargets();
```
Comment (wesleyboar) [inline taccsite_cms/templates/nav_cms.html:15]: This should not have only updated links in main content. It should also update links in header, main, and footer content.

**Review by wesleyboar — COMMENTED** [review]
Notes for Reviewers

**Review by duckonomy — APPROVED** [review]
Seems like the brainmap pprd external link still opens in the same window.
But the example external link & header & footer links do open in a new tab.

**Comment by wesleyboar** [comment]
Hm, I see [the same](https://github.com/TACC/Core-CMS/pull/548#pullrequestreview-1089309184). I revert PR back to Draft state. I will fix this blasted feature.

_**Note to Self**: The "Tools ▸ Sleuth 3.0.4" link is a relative link that does not have `target="_blank"`. Did I forget to deploy?_

File: taccsite_cms/templatetags/tacc_uri_shortcuts.py (hunk 1 above)
Comment (wesleyboar) [reply taccsite_cms/templatetags/tacc_uri_shortcuts.py:114]: This was deployed to BrainMap `pprd`, but overwritten by test deploy of [v3.8.1](https://github.com/TACC/Core-CMS/releases/tag/v3.8.1).

**Comment by wesleyboar** [comment]
@duckonomy BrainMap (pprd) should be working again now. Please confirm. (I had deployed another image for another PR atop it.)

**Review by duckonomy — APPROVED** [review]
@wesleyboar LGTM!

**Pull request merged**